### ISSUE TITLE
Writing empty files into ZipOutputStream triggers CRC32 error

### ISSUE THREAD

**Issue by ghost** [icsharpcode/SharpZipLib#123]
### Steps to reproduce
1. Create a 0-byte file on the disk.
2. Try to zip it using code like this:

```
FileStream ostream = File.OpenRead(file);
obuffer = new byte[ostream.Length];
ostream.Read(obuffer, 0, obuffer.Length);
zipOutputStream.Write(obuffer, 0, obuffer.Length);
```
### Expected behavior

The file should be added successfully to the zip stream.
### Actual behavior

A CRC32 error is thrown ("not a valid index into buffer").
### Workaround

Prevent writing into the archive stream the empty file, e.g.

```
if (obuffer.Length > 0)
    zipOutputStream.Write(obuffer, 0, obuffer.Length);
```
### Obtained from (place an x between the brackets for all that apply)
- [x] Compiled from source
  - branch: master
  - commit: 00c3c91
- [ ] Downloaded DLL from GitHub
- [ ] Downloaded DLL from SourceForge
- [ ] Downloaded DLL from _______
- [ ] DLL included as part of
- Package installed using:
  - [ ] NuGet
  - [ ] MyGet
  - [ ] Chocolatey

**Comment by Numpsy** [comment]
Think this one is already fixed, but extra unit test for the issue @ #469 in case it's useful.

**Comment by piksel** [comment]
Now we can close this with good conscience!
